### PR TITLE
Fixed italicised Units in sdec_plot

### LINKED ISSUE #1445: Remove italicised units in labels of SDEC plot in matplotlib
To remove the italicised units, you could try something along the lines of:
plt.ylabel('Luminosity (erg $\mathrm{s^{-1}}$ $\mathrm{\AA^{-1}}$)') 
This is an example where the labels won't be italicised, since they are in \mathrm{}

_Originally posted by @jamesgillanders in https://github.com/tardis-sn/tardis/pull/1442#discussion_r568030946_

### PR DESCRIPTION
**Pull request by DhruvSondhi**
This PR tries to Fix #1445 
## Description

Changed the labels to as to match the [comment](https://github.com/tardis-sn/tardis/pull/1442/files#r568030946) as mentioned
## Motivation and Context

Fixes #1445 
## How Has This Been Tested?
- [ ] Testing pipeline
- [ ] Reference Data Comparison following these [instructions](https://tardis-sn.github.io/tardis/development/continuous_integration.html)
- [x] Other (please describe)

Local Changes 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] None of the above (please describe)

## Checklist:

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have built the documentation on my fork following [these instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html)
- [ ] I have assigned and requested two reviewers for this pull request

### CODE REVIEW

**Comment by jaladh-singhal** [comment]
@DhruvSondhi thanks for this - can you show the screenshots comparing how plot looks like before & after these changes?

**Comment by DhruvSondhi** [comment]
Hello @jaladh-singhal ... I am afraid that I don't know the code where I can check this plot ... Can you please share a boiler plate test code or please guide me where I can find the same so that I can plot this graph 😄

**Comment by DhruvSondhi** [comment]
Hello, I found the way to make the plot but I am facing an issue with the quickstart data set & representation. The error seems to be something related to `Pandas` not able to pass a key in a list format.
```python
KeyError: "Passing list-likes to .loc or [] with any missing labels is no longer supported. The following labels were missing: MultiIndex([( 8, 0),\n            (12, 0),\n            (14, 0),\n            (16, 0),\n            (18, 0),\n            (20, 0)],\n           names=['atomic_number', 'ion_number']). See https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike"
```

**Comment by jaladh-singhal** [comment]
> Hello @jaladh-singhal ... I am afraid that I don't know the code where I can check this plot ... Can you please share a boilerplate test code or please guide me where I can find the same so that I can plot this graph smile

Sorry for late reply. I'm actually working on adding a demo notebook for SDEC plot, you can download and use it from [here](https://github.com/jaladh-singhal/tardis/blob/faca1511495b289d50e0794612f817e9332dc7a3/docs/using/gui/sedec_plot.ipynb)

> Hello, I found the way to make the plot but I am facing an issue with the quickstart data set & representation. The error seems to be something related to Pandas not able to pass a key in a list format. [Check this link for the error which is KeyError]
Check this Gist out ... Please do advise how I can rectify this?

@DhruvSondhi this seems strange. I hope you're using the latest tardis codebase and using development installation by `python setup.py develop`? Also if you have ran quickstart notebook before, can you compare between what is different in your development environment now?

**Comment by DhruvSondhi** [comment]
Thank you very much @jaladh-singhal for the reply 😄 ... Yes I do seem so this is a problem and I have opened an issue #1447 ... Please check this issue out :)

**Comment by DhruvSondhi** [comment]
Also, I have found some more issues which are not that important but are present so I will discuss in another issue 😉

**Comment by DhruvSondhi** [comment]
Hello @jaladh-singhal, I am afraid that the notebook you have shared is not working as intended. It does raise a peculiar error which is as follows :
```python
IndexError                                Traceback (most recent call last)
<ipython-input-7-4da046e795d2> in <module>
----> 1 plotter.generate_plot_mpl(packet_wvl_range=[3000, 9000] * u.AA);

~/Repositories/tardis/tardis/widgets/sdec_plot.py in generate_plot_mpl(self, packets_mode, packet_wvl_range, distance, show_modeled_spectrum, ax, figsize, cmapname)
    870         )
    871 
--> 872         self._show_colorbar_mpl()
    873 
    874         # Set legends and labels

~/Repositories/tardis/tardis/widgets/sdec_plot.py in _show_colorbar_mpl(self)
    967         mappable = cm.ScalarMappable(norm=norm, cmap=custcmap)
    968         mappable.set_array(np.linspace(1, self.elements.size + 1, 256))
--> 969         cbar = plt.colorbar(mappable, ax=self.ax)
    970 
    971         bounds = np.arange(self.elements.size) + 0.5

~/anaconda3/envs/tardis-2/lib/python3.7/site-packages/matplotlib/pyplot.py in colorbar(mappable, cax, ax, **kw)
   2192     if ax is None:
   2193         ax = gca()
-> 2194     ret = gcf().colorbar(mappable, cax=cax, ax=ax, **kw)
   2195     return ret
   2196 colorbar.__doc__ = matplotlib.colorbar.colorbar_doc

~/anaconda3/envs/tardis-2/lib/python3.7/site-packages/matplotlib/figure.py in colorbar(self, mappable, cax, ax, use_gridspec, **kw)
   2341                              'panchor']
   2342         cb_kw = {k: v for k, v in kw.items() if k not in NON_COLORBAR_KEYS}
-> 2343         cb = cbar.colorbar_factory(cax, mappable, **cb_kw)
   2344 
   2345         self.sca(current_ax)

~/anaconda3/envs/tardis-2/lib/python3.7/site-packages/matplotlib/colorbar.py in colorbar_factory(cax, mappable, **kwargs)
   1732         cb = ColorbarPatch(cax, mappable, **kwargs)
   1733     else:
-> 1734         cb = Colorbar(cax, mappable, **kwargs)
   1735 
   1736     cid = mappable.callbacksSM.connect('changed', cb.update_normal)

~/anaconda3/envs/tardis-2/lib/python3.7/site-packages/matplotlib/colorbar.py in __init__(self, ax, mappable, **kwargs)
   1226             if isinstance(mappable, martist.Artist):
   1227                 _add_disjoint_kwargs(kwargs, alpha=mappable.get_alpha())
-> 1228             ColorbarBase.__init__(self, ax, **kwargs)
   1229 
   1230     @cbook.deprecated("3.3", alternative="update_normal")

~/anaconda3/envs/tardis-2/lib/python3.7/site-packages/matplotlib/cbook/deprecation.py in wrapper(*args, **kwargs)
    449                 "parameter will become keyword-only %(removal)s.",
    450                 name=name, obj_type=f"parameter of {func.__name__}()")
--> 451         return func(*args, **kwargs)
    452 
    453     return wrapper

~/anaconda3/envs/tardis-2/lib/python3.7/site-packages/matplotlib/colorbar.py in __init__(self, ax, cmap, norm, alpha, values, boundaries, orientation, ticklocation, extend, spacing, ticks, format, drawedges, filled, extendfrac, extendrect, label)
    489         else:
    490             self.formatter = format  # Assume it is a Formatter or None
--> 491         self.draw_all()
    492 
    493     def _extend_lower(self):

~/anaconda3/envs/tardis-2/lib/python3.7/site-packages/matplotlib/colorbar.py in draw_all(self)
    516         self._config_axis()  # Inline it after deprecation elapses.
    517         # Configure axes limits, patch, and outline.
--> 518         xy = self._outline(X, Y)
    519         xmin, ymin = xy.min(axis=0)
    520         xmax, ymax = xy.max(axis=0)

~/anaconda3/envs/tardis-2/lib/python3.7/site-packages/matplotlib/colorbar.py in _outline(self, X, Y)
    768         N = X.shape[0]
    769         ii = [0, 1, N - 2, N - 1, 2 * N - 1, 2 * N - 2, N + 1, N, 0]
--> 770         x = X.T.reshape(-1)[ii]
    771         y = Y.T.reshape(-1)[ii]
    772         return (np.column_stack([y, x])

~/anaconda3/envs/tardis-2/lib/python3.7/site-packages/numpy/ma/core.py in __getitem__(self, indx)
   3223         # mask of being reshaped if it hasn't been set up properly yet
   3224         # So it's easier to stick to the current version
-> 3225         dout = self.data[indx]
   3226         _mask = self._mask
   3227 

IndexError: index 2 is out of bounds for axis 0 with size 2
```
I don't understand if it's something related to the virtual env or is it my Tardis installation 🤔

**Comment by DhruvSondhi** [comment]
Also, there is a minor change in the notebook 
Instead of 
```python
from tardis.widgets import SEDecPlotter
```
It should be
```python
from tardis.widgets import SDECPlotter
```
in this [notebook](https://github.com/jaladh-singhal/tardis/blob/faca1511495b289d50e0794612f817e9332dc7a3/docs/using/gui/sedec_plot.ipynb)

**Comment by DhruvSondhi** [comment]
Hello @jaladh-singhal, I got it working at last ... I do suggest a change which is that this line:
image
be at the top of the cell execution than at the bottom of the execution as it is very easy to skim over unnoticed 😄

**Comment by DhruvSondhi** [comment]
So these are the screenshots (And yes they are awesome 😉 , thanks to @jamesgillanders)
**[Luminosity]**
Before:
image
After:
image

**[Flux]**
Before:
image
After:
image

*{PS: that italics F(lambda) & L(lambda) are due to that lambda passed ... I can try to fix them too but I am not sure about them 😓 }*

**Comment by jaladh-singhal** [comment]
@DhruvSondhi regarding your suggestions of `SDECPlotter` and of `virtual_packet_logging`, good catch! They have already been accounted in #1438 since I told you I was working on this notebook - you can access them if you re-download the notebook now.

**Comment by jaladh-singhal** [comment]
> Hello @jaladh-singhal, I got it working at last ... I do suggest a change which is that this line:
> image
> be at the top of the cell execution than at the bottom of the execution as it is very easy to skim over unnoticed 😄  

@jamesgillanders @ycamacho do you think we should move it above, i.e. before `run_tardis()` cell?

**Comment by jaladh-singhal** [comment]
@DhruvSondhi the screenshots look good. @jamesgillanders can you double check that you wanted `Luminosity` instead of `L_{\lambda}` and same for flux (rest looks good to me)?

**Comment by jaladh-singhal** [comment]
**Note: @jamesgillanders don't merge it until #1438 is merged and @DhruvSondhi rebases this PR**

**Comment by DhruvSondhi** [comment]
> **Note: @jamesgillanders don't merge it until #1438 is merged and @DhruvSondhi rebases this PR**

Sure I can do that ... No issues 😄

File: tardis/widgets/sdec_plot.py
Comment (jamesgillanders) [inline tardis/widgets/sdec_plot.py:876]: this x-axis unit is still italicised

File: tardis/widgets/sdec_plot.py
Comment (jamesgillanders) [inline tardis/widgets/sdec_plot.py:880]: There seems to be some confusion about what to change on the axis labels - I didn't help things with a vague comment, but I am not proposing any change to the actual label - just that the Angstrom unit is no longer italicised. The comment posted was just grabbed from one of my programs illustrating how to use mathrm. Can this be reverted back to F_lambda?

File: tardis/widgets/sdec_plot.py
Comment (jamesgillanders) [inline tardis/widgets/sdec_plot.py:886]: Same as above - roll back to L_lambda

**Review by jamesgillanders — COMMENTED** [review]
See comments in code - the only that this PR should be changing is the italicised units in the x and y axes. Leave the F_lambda and L_lambda as is

**Comment by jamesgillanders** [comment]
@jaladh-singhal - I have no strong opinions on where exactly the virtual_packet_logging flag should appear. Whatever you think is most suitable is fine.

File: tardis/widgets/sdec_plot.py
Comment (DhruvSondhi) [reply tardis/widgets/sdec_plot.py:880]: Yes, No Issues but the F_lambda & L_lambda will be italicised ... I will update the screenshots with the same :)

File: tardis/widgets/sdec_plot.py
Comment (DhruvSondhi) [reply tardis/widgets/sdec_plot.py:886]: Sure, No problems :)

File: tardis/widgets/sdec_plot.py
Comment (DhruvSondhi) [reply tardis/widgets/sdec_plot.py:876]: Yes, I will do that too but the problem didn't seem to occur in the X-axis though I will hard code it that way :)

File: tardis/widgets/sdec_plot.py
Comment (DhruvSondhi) [reply tardis/widgets/sdec_plot.py:880]: Flux
image

File: tardis/widgets/sdec_plot.py
Comment (DhruvSondhi) [reply tardis/widgets/sdec_plot.py:886]: Luminosity
image

**Comment by DhruvSondhi** [comment]
Just pushing changes ... this looks awesome & good to go :)

**Review by jamesgillanders — APPROVED** [review]
the code modifications fix the flagged issue - happy for it to be merged, pending second review

**Comment by DhruvSondhi** [comment]
Hello @jaladh-singhal, I have rebased this branch upon the updated master ... Please check & confirm the same 😄

**Review by jaladh-singhal — APPROVED** [review]
Looks good to me!

**Pull request merged**
